tsconfig.packages.json

```diff
@@ -4,6 +4,7 @@
     "esModuleInterop": true,
     "lib": ["ES2020"],
     "module": "Node16",
+    "isolatedModules": true,
```

Comment:
Added `isolatedModules: true` to enforce proper type/value export separation. 

This ensures each file can be transpiled independently without knowledge of the full type system, which:
1. Improves build performance in bundlers/transpilers like ts-jest
2. Prevents runtime errors from improper type imports
3. Makes the codebase more compatible with modern tooling

The related changes in export syntax (`export type {...}` vs `export {...}`) ensure proper distinction between runtime values and compile-time types.